### ISSUE TITLE
Allow setting the result of a task completion source from an existing task.

### ISSUE THREAD

**Issue by richardjrossiii** [BoltsFramework/Bolts-iOS#117]
Frequently, I have to write code like this:

   BFTaskCompletionSource *source = [BFTaskCompletionSource taskCompletionSource];
    [.... continueWithBlock:^id(BFTask *task) {
        if (task.faulted) {
            NSError *error = task.error;
            if (error) {
                [source trySetError:error];
            } else {
                [source trySetException:task.exception];
            }
        } else if (task.cancelled) {
            [source trySetCancelled];
        } else {
            [source trySetResult:task.result];
        }
        
        return task;
    }];

It would be nice if there was an API similar to `[source trySetTask:task]` instead of having this ugly block of code.

**Comment by andrewtoth** [comment]
```

import Bolts

extension BFTaskCompletionSource {

    func setTask(task: BFTask) {
        if task.faulted {
            if task.error != nil {
                setError(task.error)
            } else {
                setException(task.exception)
            }
        } else if task.cancelled {
            cancel()
        } else {
            setResult(task.result)
        }
    }
}

```

**Comment by nlutsenko** [comment]
cc @grantland What do you think?
Using `updateFromTask:` on a BFTaskCompletionSource actually makes slightly more sense, since we already have a `task` aka `getTask` property on it.

**Comment by grantland** [comment]
First, when do we really need to use this pattern, how often do we need to use this pattern, and is it really something we se other developers utilizing? I'm not quite convinced this needs to be pushed upstream.

**Comment by bob9630** [comment]
This pattern is necessary when interacting with libraries that aren't designed with Bolts in mind. For example, Let's say I have an initialization task, followed by a NetworkTask where I create an NSURLSessionDataTask which has a completionHandler that has a few of sub-tasks to run which should eventually hit a FinalTask which is in charge of handling errors generated in all tasks above it as well as handling the success case if everything goes as planed.

InitTask
NetworkTask
     network-subtaskA
     network-subtaskB
FinalTask

If I didn't have a NetworkTask which had it's own completionHandler outside the scope of Bolts, I could do this simply by returning a task chain in the NetworkTask in the continueWith block which intelligently would determine the if the result returned is a BFTask handle it one way, else wrap the result in a BFTask and handle it another way. 

In short the request here is to have the same ability when setting the results of a completionSource as we do when returning results from a  continueWith block.
